### PR TITLE
Added blacklisting and searching by regexp

### PR DESCRIPTION
**Pull request by maciejmiedzianowski**

### CODE REVIEW

File: src/main/java/de/azapps/kafkabackup/common/topic/restore/RestoreArgsWrapper.java [hunk 1]
```diff
@@ -37,7 +40,8 @@
   public static final String KAFKA_BOOTSTRAP_SERVERS = "kafka.bootstrap.servers";
 
   public static final String RESTORE_DRY_RUN = "restore.dryRun";
-  public static final String RESTORE_TOPIC_LIST = "restore.topicList";
+  public static final String RESTORE_TOPIC_LIST_MODE = "restore.topicsList.mode";
```
Comment (loffek) [inline src/main/java/de/azapps/kafkabackup/common/topic/restore/RestoreArgsWrapper.java]: Hm. Why not have `restore.topicDenyList` and `restore.topicAllowList` and both take regexps? Like for burrow: https://github.com/getdreams/dreams2-burrow/blob/master/config/burrow-dreams2-stage.toml#L28-L29
And then just use both to filter? (the order they are applied shouldn't matter, unless I'm confused)

**Review by loffek — CHANGES_REQUESTED** [review]

File: src/main/java/de/azapps/kafkabackup/common/topic/restore/RestoreArgsWrapper.java (hunk 1 above)
Comment (loffek) [reply src/main/java/de/azapps/kafkabackup/common/topic/restore/RestoreArgsWrapper.java]: Just remember that an empty allowList means `.*`

Comment (maciejmiedzianowski) [reply src/main/java/de/azapps/kafkabackup/common/topic/restore/RestoreArgsWrapper.java]: right, sounds like a better solution - I'll change this!

File: src/main/java/de/azapps/kafkabackup/common/topic/restore/TopicsListMode.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package de.azapps.kafkabackup.common.topic.restore;
+
+public enum TopicsListMode {
+  ALL_TOPICS, WHITELIST, BLACKLIST, REGEXP
```
Comment (loffek) [inline src/main/java/de/azapps/kafkabackup/common/topic/restore/TopicsListMode.java]: This looks like a left-over :)

File: src/main/java/de/azapps/kafkabackup/restore/RestoreTopicService.java [hunk 3]
```diff
@@ -29,7 +31,23 @@ public void restoreTopics(RestoreArgsWrapper restoreTopicsArgsWrapper) {
...
+        .map(TopicConfiguration::getTopicName)
+        .collect(Collectors.toList());
+
+    String topicAllowRegex = restoreArgsWrapper.getTopicsAllowListRegex();
+    String topicDenyRegex = restoreArgsWrapper.getTopicsDenyListRegex();
+
+    return topicsFromConfig.stream()
+        .filter(topicName -> topicName.matches(topicAllowRegex))
+        .filter(topicName -> !topicName.matches(topicDenyRegex))
+        .collect(Collectors.toList());
```
Comment (loffek) [inline src/main/java/de/azapps/kafkabackup/restore/RestoreTopicService.java]: nice

File: src/test/java/de/azapps/kafkabackup/restore/RestoreTopicServiceTest.java [hunk 4]
```diff
@@ -108,32 +113,41 @@ public void shouldRestoreOnlyTopicFromConfiguration() {
   }
 
   @Test
-  public void shouldNotCallCreateTopicsOnKafkaWhenInDryRun() {
+  public void shouldOmitTopicsFromBlackListDuringRestoration() {
```
Comment (loffek) [inline src/test/java/de/azapps/kafkabackup/restore/RestoreTopicServiceTest.java]: => DenyList

**Review by loffek — CHANGES_REQUESTED** [review]
Nice! Just some 💅

File: src/main/java/de/azapps/kafkabackup/restore/message/RestoreMessageService.java [hunk 5]
```diff
@@ -46,7 +47,7 @@ public void restoreMessages() {
         restoreArgsWrapper.getConfigBackupBucket());
 
     List<TopicPartitionToRestore> partitionsToRestore = getPartitionsToRestore(topicsConfig,
-        restoreArgsWrapper.getTopicsToRestore());
+        RestoreTopicService.getTopicsList(restoreArgsWrapper, topicsConfig));
```
Comment (MacRudDNA) [inline src/main/java/de/azapps/kafkabackup/restore/message/RestoreMessageService.java:50]: Nice, we're using same logic

**Review by MacRudDNA — APPROVED** [review]

**Review by loffek — APPROVED** [review]
👌

**Pull request merged**